### PR TITLE
HTTP status 499 instead of 502's for 'Context canceled' 'errors'

### PR DESCRIPTION
**Pull request by Ranger-X**
### 1. What does this change do, exactly?
Replaces (not an) HTTP error code 502 'Context canceled' with the 499 'Context canceled' as suggested in https://stackoverflow.com/questions/46234679/what-is-the-correct-http-status-code-for-a-cancelled-request

### 2. Please link to the relevant issues.
https://github.com/mholt/caddy/issues/1925
https://github.com/mholt/caddy/issues/1828
https://caddy.community/t/error-context-canceled/2034/9
and others

### 3. Which documentation changes (if any) need to be made because of this PR?
None

### 4. Checklist

- [X] I have written tests and verified that they fail without my change
- [ ] I have squashed any insignificant commits
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code
- [X] I am willing to help maintain this change if there are issues with it later

### CODE REVIEW

**Comment by francislavoie** [comment]
@mholt This looks pretty safe, IMO just needs a note in the proxy docs about this behaviour. See the TODO added, where should that const go, if at all?

File: caddyhttp/proxy/proxy.go [hunk 1]
```diff
@@ -255,6 +255,11 @@ func (p Proxy) ServeHTTP(w http.ResponseWriter, r *http.Request) (int, error) {
 			return http.StatusRequestEntityTooLarge, backendErr
 		}
 
+		// TODO: write an 499 HTTP status as constant in someplace (and also change in proxy_test.go on line: expectedStatus, expectErr := 499, context.Canceled)
```
Comment (mholt) [inline caddyhttp/proxy/proxy.go]: I guess we can start by putting the `const StatusContextCancelled = 499` at the bottom of this file

Comment (Ranger-X) [reply caddyhttp/proxy/proxy.go]: Ok. But I suggest a `CustomStatusContextCancelled` because this is not a standard HTTP status. Are you agreed?

Comment (Ranger-X) [reply caddyhttp/proxy/proxy.go]: Done.

Comment (tobya) [reply caddyhttp/proxy/proxy.go]: @mholt  Are you happy with this change.  Can you approve and I will merge.  @francislavoie would you like to approve changes also.

**Comment by mholt** [comment]
Appveyor failure is unrelated. Just waiting for the CLA to be signed / verified.

**Comment by Ranger-X** [comment]
I think I have already signed CLA. @mholt could you to recheck PRs? (https://github.com/cla-assistant/cla-assistant/issues/124)

**Comment by mholt** [comment]
Got it, thank you!

**Pull request merged**